### PR TITLE
add a metric print method

### LINKED ISSUE #435: Printing method for metrics
- I think a print method for the `metric` functions, if people are going to be creating them and dealing with them:

``` r
yardstick::sens
#> function (data, ...) 
#> {
#>     UseMethod("sens")
#> }
#> <bytecode: 0x117ee7e30>
#> <environment: namespace:yardstick>
#> attr(,"direction")
#> [1] "maximize"
#> attr(,"class")
#> [1] "class_metric" "metric"       "function"

## just an idea:
format.metric <- function(x, ...) {
    first_class <- class(x)[[1]]
    cli::cli_format_method({
        cli::cli_h3("A {.cls {first_class}} function to {attr(x, 'direction')} metrics")
    })
}

print.metric <- function(x, ...) {
    cat(format(x), sep = "\n")
    invisible(x)
}

yardstick::sens
#> 
#> ── A <class_metric> function to maximize metrics
```

<sup>Created on 2023-05-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

_Originally posted by @juliasilge in https://github.com/tidymodels/yardstick/pull/434#pullrequestreview-1427157891_

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #435!

### CODE REVIEW

**Comment by EmilHvitfeldt** [comment]
As a general comment, i think we should overhaul the printing of `metric_set()` at the same time 😄

**Comment by simonpcouch** [comment]
Nice! Related to #454, then. How would you like to move this PR forward/not?

I'd be glad to work on the metric set method, too. I guess a one-line-output metric print method would make the metric set print method nice and consistent. :)

**Comment by EmilHvitfeldt** [comment]
I think we should add the metric set printing to this PR. we might learn things when updating that, that informs the individual printing method, so keeping them together seems easier

**Comment by simonpcouch** [comment]
I'm thinking this could look like:

``` r
roc_auc
#> A probability metric | direction: maximize

demographic_parity(boop)
#> A class metric       | direction: minimize, group-wise on: boop
```

...with:

``` r
metric_set(roc_auc, demographic_parity(boop))
#> A metric set, consisting of:
#> - A probability metric | direction: maximize
#> - A class metric       | direction: minimize, group-wise on: boop
```

...with cli formatting where needed. Notably, I think it'd be great to link `probability metric` and `class metric` to a `{.help}`-page clarifying what that means, if possible.

Thoughts?

**Comment by EmilHvitfeldt** [comment]
I like the look of that!

I would like to have the names of the metrics in the metric set printing.

**Comment by simonpcouch** [comment]
Spent some time on this yesterday but didn't quite have time to finish—moving on to other things for at least today. Still need to write code for justification (see `rlib/cli#229`).

**Comment by simonpcouch** [comment]
Currently, output looks like:

``` r
library(yardstick)
dem_parity_boop <- demographic_parity(boop)

roc_auc 
#> A probability metric | direction: maximize

dem_parity_boop
#> A class metric | direction: minimize, group-wise on: boop

demographic_parity(boop)
#> A class metric | direction: minimize, group-wise on: boop

metric_set(roc_auc, dem_parity_boop)
#> A metric set, consisting of:
#> - `roc_auc()`, a probability metric   | direction: maximize
#> - `dem_parity_boop()`, a class metric | direction: minimize, group-wise on:
#> boop

metric_set(roc_auc, demographic_parity(boop))
#> A metric set, consisting of:
#> - `roc_auc()`, a probability metric            | direction: maximize
#> - `demographic_parity(boop)()`, a class metric | direction: minimize,
#> group-wise on: boop
```

<sup>Created on 2023-11-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

While `format.metric_set()` _does_ call `format.metric`, there's a good bit of post-processing that needs to happen to make justification work. From my reading of cli issues, we don't have many justification features to draw from there. I generally do feel like the rows x columns appearance for printing works well for metric sets and miss that structure with my current attempt. I may hand this off to you here, @EmilHvitfeldt, either for review or revision.

File: NEWS.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ calculated with `roc_auc_survival()`.
 
 * `demographic_parity()`, `equalized_odds()`, and `equal_opportunity()` are new metrics for measuring model fairness. Each is implemented with the `new_groupwise_metric()` constructor, a general interface for defining group-aware metrics that allows for quickly and flexibly defining fairness metrics with the problem context in mind.
 
+* Added a print method for metrics (#435).
```
Comment (EmilHvitfeldt) [inline NEWS.md:19]:
```suggestion
* Added a print method for metrics and metric sets (#435).
```

**Comment by EmilHvitfeldt** [comment]
This all looks good! added a couple of small changes, will merge when CI runs clean

**Pull request merged**